### PR TITLE
Config option - injecting document.domain

### PR DESCRIPTION
**Pull request by cacieprins**
(draft pr for work relating to https://github.com/cypress-io/cypress/issues/29590)

### CODE REVIEW

File: docs/app/references/configuration.mdx [hunk 1]
```diff
@@ -588,6 +589,55 @@ The `**/node_modules/**` pattern is automatically added to `excludeSpecPattern`
 and does not need to be specified (and can't be overridden). See [`e2e`](#e2e)
 or [`component`](#component) testing-specific options.
 
+### injectDocumentDomain
+
+This option is <Badge type="caution">deprecated</Badge>, and will be removed in the next
+major version of Cypress.
+
+Set this configuration option is `true` to instruct Cypress to
```
Comment (jennifer-shehane) [inline docs/app/references/configuration.mdx:597]:
```suggestion
Set this configuration option to `true` to instruct Cypress to
```

File: docs/app/references/configuration.mdx [hunk 2]
```diff
@@ -588,6 +589,55 @@ The `**/node_modules/**` pattern is automatically added to `excludeSpecPattern`
...
+This configuration value _must_ be set to true if you are running tests with experimental
+WebKit support, as `cy.origin` is not yet supported in WebKit.
+
+:::caution
+
+#### Known Incompatibilities
+
+Setting this configuration value to `true` can cause the application you are testing
+to behave in unexpected ways. This is especially true if a site your tests visit
+sets the [`Origin-Agent-Cluster`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin-Agent-Cluster)
```
Comment (jennifer-shehane) [inline docs/app/references/configuration.mdx:617]:
```suggestion
to behave in unexpected ways. This is especially true if the site under test
sets the [`Origin-Agent-Cluster`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin-Agent-Cluster)
```

File: docs/app/references/migration-guide.mdx [hunk 3]
```diff
@@ -68,6 +68,44 @@ For users with the existing `experimentalJustInTimeCompile` flag set, you can re
 
 :::
 
+### Changes to `cy.origin()`
+
+To support sites that use [origin-keyed agent clusters](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin-Agent-Cluster), Cypress no longer injects `document.domain` into `text/html` content by default.
```
Comment (jennifer-shehane) [inline docs/app/references/migration-guide.mdx:73]: I'd like to mention the document.domain deprecation somewhere here: https://developer.chrome.com/blog/document-domain-setter-deprecation - just want to give context that these are outward changes enforcing this direction.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]

File: docs/app/references/experiments.mdx [hunk 4]
```diff
@@ -257,63 +257,10 @@ creating `e2e` and `component` objects inside your Cypress configuration.
...
-
-:::
-
-If using other Salesforce domains, such as
-[enhanced domains](https://help.salesforce.com/s/articleView?id=sf.domain_name_enhanced.htm&type=5),
-you will need to add the correct matching glob pattern.
+| Option                    | Default | Description                                                                                               |
+| ------------------------- | ------- | --------------------------------------------------------------------------------------------------------- |
+| `experimentalStudio`      | `false` | Generate and save commands directly to your test suite by interacting with your app as an end user would. |
+| `experimentalRunAllSpecs` | `false` | Enables the "Run All Specs" UI feature, allowing the execution of multiple specs sequentially.            |
```
Comment (jennifer-shehane) [inline docs/app/references/experiments.mdx:263]: @cacieprins Did you mean to remove `experimentalOriginDependencies` here?

Comment (cacieprins) [reply docs/app/references/experiments.mdx:263]: Accidental - resolved in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/references/migration-guide.mdx (hunk 3 above)
Comment (cacieprins) [reply docs/app/references/migration-guide.mdx]: Addressed in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/references/configuration.mdx (hunk 2 above)
Comment (cacieprins) [reply docs/app/references/configuration.mdx:618]: addressed in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/references/configuration.mdx (hunk 1 above)
Comment (cacieprins) [reply docs/app/references/configuration.mdx]: addressed in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/guides/cross-origin-testing.mdx [hunk 5]
```diff
@@ -136,38 +122,33 @@ chart to help clarify the differences!
...
+two different origins in the same test and continue to interact with the page without
+the use of the [`cy.origin()`](/api/commands/origin) command.
 
 :::
 
 :::tip
 
 <Icon name="check-circle" color="green" /> You **can** [visit](/api/commands/visit)
-two or more domains of different origin in **different** tests without needing [`cy.origin()`](/api/commands/origin).
+two or origins in **different** tests without needing [`cy.origin()`](/api/commands/origin).
```
[the remark this answers is not in the document]
Comment (cacieprins) [reply docs/app/guides/cross-origin-testing.mdx]: addressed in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/guides/cross-origin-testing.mdx [hunk 6]
```diff
@@ -72,38 +69,38 @@ It's important to note that although we do our **very best** to ensure your
...
 information.
 
 #### Examples of test cases that will error without the use of `cy.origin`
 
 1. [`.click()`](/api/commands/click) an `<a>` with an `href` to a different
-   superdomain with subsequent Cypress commands being run.
+   origin with subsequent Cypress commands being run.
 2. [`.submit()`](/api/commands/submit) a `<form>` that causes your web server to
-   redirect you a different superdomain where additional Cypress commands are
+   redirect you a different origin where additional Cypress commands are
```
[the remark this answers is not in the document]
Comment (cacieprins) [reply docs/app/guides/cross-origin-testing.mdx]: addressed in https://github.com/cypress-io/cypress-documentation/pull/5974/commits/644aa612b9a136472c6f1be955f49458f3770b70

File: docs/app/guides/cross-origin-testing.mdx [hunk 7]
```diff
@@ -113,17 +108,6 @@ used.
 Read on to learn about
 [working around these common problems](#Common-Workarounds).
 
-#### What is a superdomain?
-
-But what is same superdomain? It's actually very similar to that of same
```
Comment (AtofStryker) [inline docs/app/guides/cross-origin-testing.mdx:118]: yay to superdomain no more!

File: docs/app/guides/cross-origin-testing.mdx [hunk 8]
```diff
@@ -136,38 +120,33 @@ chart to help clarify the differences!
...
-`https://www.cypress.io`.
-
-- `https://cypress.io`
-- `https://docs.cypress.io`
-- `https://example.cypress.io/commands/querying`
+##### Origin
 
-The URLs below, however, will have different superdomains/origins compared to
-`https://www.cypress.io`.
+An `origin` is comprised of a URL's `scheme`, `hostname`, _and_ `port`. Given the URLs
```
Comment (AtofStryker) [inline docs/app/guides/cross-origin-testing.mdx:125]: this is so much easier to interpret now as a normal web user IMO

File: docs/app/references/configuration.mdx [hunk 9]
```diff
@@ -589,6 +590,55 @@ The `**/node_modules/**` pattern is automatically added to `excludeSpecPattern`
 and does not need to be specified (and can't be overridden). See [`e2e`](#e2e)
 or [`component`](#component) testing-specific options.
 
+### injectDocumentDomain
+
+This option is <Badge type="caution">deprecated</Badge>, and will be removed in the next
+major version of Cypress.
```
Comment (AtofStryker) [inline docs/app/references/configuration.mdx:596]: Do we want to be more explicit here?
```suggestion
This option is <Badge type="caution">deprecated</Badge>, and will be removed in Cypress. 15
```

File: docs/app/references/configuration.mdx [hunk 10]
```diff
@@ -589,6 +590,55 @@ The `**/node_modules/**` pattern is automatically added to `excludeSpecPattern`
...
+
+This configuration value _must_ be set to true if you are running tests with experimental
+WebKit support, as `cy.origin` is not yet supported in WebKit.
+
+:::caution
+
+#### Known Incompatibilities
+
+Setting this configuration value to `true` can cause the application you are testing
+to behave in unexpected ways. This is especially true if a site your tests visit
```
Comment (AtofStryker) [inline docs/app/references/configuration.mdx:617]: there are implications if your site sets it, but there are also implications where the browser defaults the `origin-agent-cluster` into buckets, which can cause problems. I think this is also worth mentioning

File: docs/api/commands/origin.mdx [hunk 11]
```diff
@@ -64,7 +76,7 @@ const hits = getHits()
 cy.visit('https://example.cypress.io/history/founder')
 // To interact with cross-origin content, move this inside cy.origin() callback
 cy.get('h1').contains('About our Founder')
-// Domain must be a precise match including subdomain, i.e. example.cypress.io
+// Origin must be a precise match including scheme, subdomain and port, i.e. https://example.cypress.io
 cy.origin('cypress.io', () => {
```
[the remark this answers is not in the document]
Comment (jennifer-shehane) [reply docs/api/commands/origin.mdx]: Addressed here: [`c2510e9` (#5974)](https://github.com/cypress-io/cypress-documentation/pull/5974/commits/c2510e9eb7c8edc7f46069117f9d0638fa756008)

**Comment by jennifer-shehane** [comment]
I have completely messed up this branch and don't even know how. 🙏🏻 Opening https://github.com/cypress-io/cypress-documentation/pull/6058 to replace it.

**Pull request closed without merging**